core/container.go

```diff
@@ -356,8 +356,11 @@ func (container *Container) From(ctx context.Context, gw bkgw.Client, addr strin
 	// associate vertexes to the 'from' sub-pipeline
 	recordVertexes(subRecorder, container.FS)
 
-	// merge config.Env with imgSpec.Config.Env
-	imgSpec.Config.Env = append(container.Config.Env, imgSpec.Config.Env...)
+	// Merge environment, labels and exposed ports from previous container to the new one.
+	// If the environment variable, label or exposed port also exists in the new configuration,
+	// it will replace the old one.
+	imgSpec.Config = mergeImageConfig(container.Config, imgSpec.Config)
+
 	container.Config = imgSpec.Config
```

Comment:
We want to apply the config from the pulled image (`imgSpec`) **on top** of the existing `container` config:
```suggestion
	container.Config = mergeImageConfig(imgSpec.Config, container.Config)
```
Which explains the reversed arguments in `mergeImageConfig`.

Reply:
Fixed in [32f5bad](https://github.com/dagger/dagger/pull/5052/commits/32f5bad50146e1aedd101f6a347c1009de69ebf7)

Reply:
You missed removing `container.Config = imgSpec.Config`.